### DISCUSSION TITLE
Selecting entities in package creation, should include all dependencies

### DISCUSSION THREAD

**Discussion by skttl** [umbraco/Umbraco-CMS#13668] [Features and ideas]
### Which *exact* Umbraco version are you using? For example: 9.0.1 - don't just write v9

9.0.1

### Bug summary

When creating a package, you have to manually add all dependencies of the different entities you select. So as an example, if I pick a doc type, and download the package.xml, it will only contain the document type. None of the data types used in the document type is added.

It would be nice if it could resolve and add all dependencies of the document type, including data types, compositions, templates etc.

Also if I add content, it should automatically add the document type of the node including all its dependencies.

### Steps to reproduce

- Go to the package section
- Create a package
- Select a content node

### Expected result / actual result

I would expect a package.xml file containing not only the xml for the selected content node, but also the nodes document type, the document types data types etc.

**Comment by nul800sebastiaan** [comment]
Thanks @skttl - definitely an idea for the future, a lot of work though if you want to start doing dependency resolving, then how about the template on the doctype, selected media items, the media type etc.. :)

**Comment by skttl** [comment]
Gave this some thoughts - it could also work by having the UI showing the user which things could be necessary to include.

See below, when I add Contact, it will query the backend for which document type and template it is made of, and add a help text below that one.
![2022-07-15-13-57-52-940__twjqOu0VG6](https://user-images.githubusercontent.com/3726467/179219261-3f7e211d-9db1-416f-a9c6-f725733ffb16.png)

I then add the Contact document type, and it will query again, revealing some necessary composition types (and further below data types, templates etc.)
![2022-07-15-13-59-43-984__iMWozrlriH](https://user-images.githubusercontent.com/3726467/179219267-a72e965a-b49b-4da1-bc43-d0bfbee5f031.png)

**Comment by kjac** [comment]
Hi @skttl 👋 

I'm going to move this to the GitHub discussions as this is really a feature request. Perhaps that will attract a little more attention to this idea 😄
